File: Python-Program/Q1.py
class BankAccount:
    def _init_(self, owner, balance=0):
        self.owner = owner
        self.balance = balance

    def deposit(self, amount):
        if amount <= 0:
            print("Deposit amount must be positive.")
        else:
            self.balance += amount
            print(f"Deposited: ${amount}")

    def withdraw(self, amount):
        if amount > self.balance:
            print("Insufficient funds.")
        elif amount <= 0:
            print("Withdrawal amount must be positive.")
        else:
            self.balance -= amount
            print(f"Withdrew: ${amount}")

    def get_balance():
        return self.balance

# Program starts here
account = BankAccount("Alice", 100)
print("Welcome,", account.owner)

while True:
    action = input("Choose action (deposit, withdraw, balance, exit): ").lower()
    if action == "deposit":
        amt = float(input("Enter amount to deposit: "))
        account.deposit(amt)
    elif action == "withdraw":
        amt = float(input("Enter amount to withdraw: "))
        account.withdraw(amt)
    elif action == "balance":
        print("Current balance: $", account.get_balance)
    elif action == "exit":
        print("Thank you for banking with us.")
        break
    else
        print("Invalid action. Try again.")

File: java-Programs/Q1.java
import java.util.Scanner;

public class BankAccount {
    private String owner;
    private double balance;

    public BankAccount(String owner, double balance) {
        this.owner = owner;
        this.balance = balance;
    }

    public void deposit(double amount) {
        if (amount <= 0) 
            System.out.println("Deposit amount must be positive.");
        else {
            balance += amount;
            System.out.println("Deposited: $" + amount);
        }
    }

    public void withdraw(double amount) {
        if (amount > balance) {
            System.out.println("Insufficient funds.");
        } else if (amount <= 0) 
            System.out.println("Withdrawal amount must be positive.");
        else
            balance -= amount;
            System.out.println("Withdrew: $" + amount); // Logical error: bracket missing above
    }

    public double getBalance {
        return balance; // Syntax error: missing parentheses in method declaration
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        BankAccount account = new BankAccount("Alice", 100);

        System.out.println("Welcome, " + account.owner); // Access modifier issue: owner is private

        while (true) {
            System.out.print("Choose action (deposit, withdraw, balance, exit): ");
            String action = sc.nextLine().toLowerCase();

            if (action.equals("deposit")) {
                System.out.print("Enter amount to deposit: ");
                double amt = sc.nextDouble();
                account.deposit(amt);
            } else if (action.equals("withdraw")) {
                System.out.print("Enter amount to withdraw: ");
                double amt = sc.nextDouble();
                account.withdraw(amt);
            } else if (action.equals("balance")) {
                System.out.println("Current balance: $" + account.getBalance); // Missing parentheses
            } else if (action.equals("exit")) {
                System.out.println("Thank you for banking with us.");
                break;
            } else
                System.out.println("Invalid action. Try again.");
        }

        sc.close();
    }
}